content/courses/Guidelines_for_Safe_and_Secure_Ada_SPARK/chapters/guidelines/concurrency/con03_avoid_shared_variables_for_inter-task_communication.rst

```diff
@@ -0,0 +1,74 @@
+-------------------------------------------------------------
+Avoid Shared Variables for Inter-task Communication (CON03)
+-------------------------------------------------------------
+
+.. include:: ../../../global.txt
+
+**Level** :math:`\rightarrow` Advisory
+
+Category
+   :Safety: :math:`\checkmark`
+   :Cyber: :math:`\checkmark`
+
+Goal
+   :Maintainability: :math:`\checkmark`
+   :Reliability: :math:`\checkmark`
+   :Portability: :math:`\checkmark`
+   :Performance: :math:`\checkmark`
+   :Security:
+
+**Remediation** :math:`\rightarrow` High
+
+**Verification Method** :math:`\rightarrow` GNATcheck rule:
+:rule:`Volatile_Objects_Without_Address_Clauses`
+
++++++++++++
+Reference
++++++++++++
+
+:arm:`D.13 The Ravenscar Profile <D-13>`
+
++++++++++++++
+Description
++++++++++++++
+
+Although the Ravenscar and Jorvik profiles allow the use of shared variables
+for inter-task communication, such use is less robust and less reliable than
+encapsulating shared variables within protected objects.
+
+++++++++++++++++++++++++++++++++++++++++++++++++
+Applicable Vulnerability within ISO TR 24772-2
+++++++++++++++++++++++++++++++++++++++++++++++++
+
+* 6.56 Undefined behaviour [EWF]
+
++++++++++++++++++++++++++++
+Noncompliant Code Example
++++++++++++++++++++++++++++
+
+.. literalinclude:: examples/con03.ads
+  :language: Ada
+  :lines: 5-7
+  :dedent: 3
+
+Note that variables marked as :ada:`Atomic` are also :ada:`Volatile`, per the
+:arm:`C.6\\/8\\(3\\) Shared Variable Control <C-6>`
+
```

Comment:
the link looks like below ( please ignore the comment if it is intended)
![Screenshot from 2023-11-16 16-24-12](https://github.com/AdaCore/learn/assets/22798363/09c65ca4-c02d-4bf9-89a6-a291c81d484a)